src/read.c

```diff
@@ -708,6 +708,30 @@ static avifResult avifDecoderItemMaxExtent(const avifDecoderItem * item, avifExt
     return AVIF_RESULT_OK;
 }
 
+static avifResult avifDecoderItemValidateAV1(const avifDecoderItem * item)
+{
+    const avifProperty * av1CProp = avifPropertyArrayFind(&item->properties, "av1C");
+    if (!av1CProp) {
+        // An av1C box is mandatory in all valid AVIF configurations. Bail out.
+        return AVIF_RESULT_BMFF_PARSE_FAILED;
+    }
+    const uint32_t av1CDepth = avifCodecConfigurationBoxGetDepth(&av1CProp->u.av1C);
+
+    const avifProperty * pixiProp = avifPropertyArrayFind(&item->properties, "pixi");
+    if (!pixiProp) {
+        // A pixi box is mandatory in all valid AVIF configurations. Bail out.
+        return AVIF_RESULT_BMFF_PARSE_FAILED;
+    }
+
+    for (uint8_t i = 0; i < pixiProp->u.pixi.planeCount; ++i) {
```

Comment:
We should first validate pixiProp->u.pixi.planeCount.

pixiProp->u.pixi.planeCount should be equal to av1CProp->u.av1C.monochrome ? 1 : 3.

Reply:
The wording on pixi (HEIF 6.5.6.3) is:

> num_channels: This field signals the number of channels by each pixel of the reconstructed image of
the associated image item.

For a long time, the primary encoders out there (libaom & rav1e) were incapable of encoding proper YUV400 for various reasons, so it had to fall back to encoding YUV420, even if it was only going to use the Y plane (such as an alpha plane).

For `av1C`, the bits of this structure are clear: they should match the actual encoded data as the intent of this box is to initialize hardware encoders. However, if the "reconstructed image" is an alpha plane (say), should the two meaningless chroma planes that come out of the AV1 decoder be mentioned in the `pixi` block?

Maybe it is more of a philosophical question - Does a decoded ("reconstructed") alpha plane that was encoded as YUV420 now have 3 channels? Or is it still just an alpha plane?

I'm willing to be convinced otherwise (and I haven't tested it), but I suspect this would be a pretty heavy breaking change for alpha planes when they were encoded with semi-incomplete AV1 encoders.

Reply:
Ah, I forgot that problem. Maybe just check the color image? Or just require pixiProp->u.pixi.planeCount is 1 or 3?